exonum-java-binding/core/rust/src/storage/proof_map_index.rs

```diff
@@ -12,49 +12,74 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use jni::{
-    objects::{JClass, JObject, JString},
-    sys::{jboolean, jbyteArray, jobject, jsize},
-    JNIEnv,
-};
-
-use std::{panic, ptr};
-
-use exonum::crypto::Hash;
 use exonum_merkledb::{
-    access::FromAccess,
+    access::{FromAccess, RawAccess},
     proof_map_index::{
-        MapProof, ProofMapIndexIter, ProofMapIndexKeys, ProofMapIndexValues, ProofPath,
-        PROOF_MAP_KEY_SIZE,
+        ProofMapIndexIter, ProofMapIndexKeys, ProofMapIndexValues, PROOF_MAP_KEY_SIZE,
     },
-    Fork, IndexAddress, ObjectHash, ProofMapIndex, Snapshot,
+    Fork, IndexAddress, ObjectHash, ProofMapIndex, RawProofMapIndex, Snapshot,
 };
-
+use exonum_proto::ProtobufConvert;
 use handle::{self, Handle};
+use jni::{
+    objects::{GlobalRef, JClass, JMethodID, JObject, JString},
+    sys::{jboolean, jbyteArray, jobject, jobjectArray, JNI_FALSE},
+    JNIEnv,
+};
+use protobuf::Message;
 use storage::{
-    db::{Value, View, ViewRef},
+    db::{Key, Value, View, ViewRef},
     PairIter,
 };
 use utils;
 use JniResult;
 
-type Key = [u8; PROOF_MAP_KEY_SIZE];
-type Index<T> = ProofMapIndex<T, Key, Value>;
+use std::{panic, ptr};
+
+type RawKey = [u8; 32];
+
+enum Index<T>
```

Comment:
I'd add a comment explaining differences between RawProofMapIndex and ProofMapIndex